code/datums/progressbar.dm

```diff
@@ -0,0 +1,226 @@
+/datum/progressbar
+	var/goal = 1
+	var/bar_tag = PROG_BAR_GENERIC
+	var/bg_tag = PROG_BG_GENERIC
+	var/frame_tag = PROG_FRAME_GENERIC
+	var/image/progress/bar/bar
+	var/image/progress/bg/bg
+	var/image/progress/frame/frame
+	var/datum/progressicon/prog_display
+	var/shown = FALSE
+	var/mob/user
+	var/client/client
+	var/listindex
+
+/datum/progressbar/New(mob/U, goal_number, atom/target, image/progress/display/new_display)
```

Comment:
```suggestion
/datum/progressbar/Initialize(mob/U, goal_number, atom/target, image/progress/display/new_display)
```

Reply:
This shouldn't need Initialize, it's merely visual and not really interactable.